### PR TITLE
Feature/md

### PR DESCRIPTION
**Pull request by Shubhangi-cs**

### CODE REVIEW

File: docs/Salesforce-batchsource.md [hunk 1]
```diff
@@ -12,14 +12,18 @@ or using sObject and incremental or range date filters.
 Configuration
 -------------
 
+**Connect Timeout:** Maximum time in milliseconds to wait for connection initialization before time out.
```
Comment (robinrielley) [inline docs/Salesforce-batchsource.md]: can you say "before it times out"?

File: docs/Salesforce-connector.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+**Consumer Key:** Application Consumer Key. This is also known as the OAuth client id.
+A Salesforce connected application must be created in order to get a consumer key.
+
+**Consumer Secret:** Application Consumer Secret. This is also known as the OAuth client secret.
+A Salesforce connected application must be created in order to get a client secret.
+
+**Login Url:** Salesforce OAuth2 login url.
+
+**Connect Timeout:** Maximum time in milliseconds to wait for connection initialization before time out.
```
Comment (robinrielley) [inline docs/Salesforce-connector.md]: can you say "before it times out"?

File: docs/Salesforce-connector.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Path of the connection
+----------------------
+To browse, get a sample from, or get the specification for this connection. (Not supported in Streaming source and
```
Comment (robinrielley) [inline docs/Salesforce-connector.md]: Can you say: (Not supported in the Salesforce Streaming source and Salesforce Multi Object batch source.)

File: docs/Salesforce-batchsink.md [hunk 4]
```diff
@@ -11,14 +11,19 @@ Currently, only inserts are supported. Upserts are not supported.
...
+
 **Reference Name:** Name used to uniquely identify this sink for lineage, annotating metadata, etc.
 
 **Username:** Salesforce username.
 
 **Password:** Salesforce password.
 
-**Security Token:** Salesforce security Token. If the password does not contain the security token the plugin 
-will append the token before authenticating with salesforce.
+**Security Token:** Salesforce security Token. If the password does not contain the security token, the plugin
```
Comment (robinrielley) [inline docs/Salesforce-batchsink.md]: Salesforce docs say "Salesforce Security Token"  We should probably match them

File: docs/Salesforce-batchsink.md [hunk 5]
```diff
@@ -72,28 +77,29 @@ Below is a non-comprehensive list of **sObjects** that are not currently availab
 - WorkOrderLineItemStatus
 - WorkOrderStatus
 
-**Operation:** Operation used for writing data into Salesforce.<br>
-Insert - adds records.<br>
+**Operation:** Operation used for writing data into Salesforce.  
+Insert - adds records.  
 Upsert - upserts the records. Salesforce will decide if sObjects 
-are the same using external id field.<br>
+are the same using external id field.
```
Comment (robinrielley) [inline docs/Salesforce-batchsink.md]: Change id to ID

File: docs/Salesforce-batchsink.md [hunk 6]
```diff
@@ -11,14 +11,19 @@ Currently, only inserts are supported. Upserts are not supported.
...
 **Username:** Salesforce username.
 
 **Password:** Salesforce password.
 
-**Security Token:** Salesforce security Token. If the password does not contain the security token the plugin 
-will append the token before authenticating with salesforce.
+**Security Token:** Salesforce security Token. If the password does not contain the security token, the plugin 
+will append the token before authenticating with Salesforce.
 
 **Consumer Key:** Application Consumer Key. This is also known as the OAuth client id.
```
Comment (robinrielley) [inline docs/Salesforce-batchsink.md]: change id to ID

File: docs/Salesforce-connector.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+**Security Token:** Salesforce security Token. If the password does not contain the security token, the plugin
+will append the token before authenticating with Salesforce.
+
+**Consumer Key:** Application Consumer Key. This is also known as the OAuth client id.
+A Salesforce connected application must be created in order to get a consumer key.
+
+**Consumer Secret:** Application Consumer Secret. This is also known as the OAuth client secret.
+A Salesforce connected application must be created in order to get a client secret.
+
+**Login Url:** Salesforce OAuth2 login url.
```
Comment (robinrielley) [inline docs/Salesforce-connector.md]: change Url and url to URL

File: docs/Salesforce-streamingsource.md [hunk 8]
```diff
@@ -19,14 +19,18 @@ The plugin can either create a PushTopic for the user, or use an existing one.
...
+
 **Reference Name:** Name used to uniquely identify this source for lineage, annotating metadata, etc.
 
 **Username:** Salesforce username.
 
 **Password:** Salesforce password.
 
-**Security Token:** Salesforce security Token. If the password does not contain the security token the plugin 
-will append the token before authenticating with salesforce.
+**Security Token:** Salesforce security Token. If the password does not contain the security token, the plugin
```
Comment (robinrielley) [inline docs/Salesforce-streamingsource.md]: change Salesforce security Token to Salesforce security token

File: docs/Salesforce-streamingsource.md [hunk 9]
```diff
@@ -19,14 +19,18 @@ The plugin can either create a PushTopic for the user, or use an existing one.
...
 **Username:** Salesforce username.
 
 **Password:** Salesforce password.
 
-**Security Token:** Salesforce security Token. If the password does not contain the security token the plugin 
-will append the token before authenticating with salesforce.
+**Security Token:** Salesforce security Token. If the password does not contain the security token, the plugin 
+will append the token before authenticating with Salesforce.
 
 **Consumer Key:** Application Consumer Key. This is also known as the OAuth client id.
```
Comment (robinrielley) [inline docs/Salesforce-streamingsource.md]: Change id to ID

File: docs/Salesforce-streamingsource.md [hunk 10]
```diff
@@ -56,14 +60,14 @@ This field not required if you are using an existing push topic.
 
 
 **Notify For Fields:** Push topic property, which specifies how the record is evaluated against the
-PushTopic query. The NotifyForFields values are:<br>
+PushTopic query. The NotifyForFields values are:
```
Comment (robinrielley) [inline docs/Salesforce-streamingsource.md]: Change NotifyForFields to Notify For Fields

File: docs/SalesforceMultiObjects-batchsource.md [hunk 11]
```diff
@@ -12,14 +12,18 @@ and the value is the schema of the SObject.
...
+
 **Reference Name:** Name used to uniquely identify this source for lineage, annotating metadata, etc.
 
 **Username:** Salesforce username.
 
 **Password:** Salesforce password.
 
-**Security Token:** Salesforce security Token. If the password does not contain the security token the plugin 
-will append the token before authenticating with salesforce.
+**Security Token:** Salesforce security Token. If the password does not contain the security token, the plugin
```
Comment (robinrielley) [inline docs/SalesforceMultiObjects-batchsource.md]: change Salesforce security token to Salesforce security token

File: docs/Salesforce-batchsource.md (hunk 1 above)
Comment (Shubhangi-cs) [reply docs/Salesforce-batchsource.md]: done

File: docs/Salesforce-connector.md (hunk 2 above)
Comment (Shubhangi-cs) [reply docs/Salesforce-connector.md]: done

File: docs/Salesforce-connector.md (hunk 3 above)
Comment (Shubhangi-cs) [reply docs/Salesforce-connector.md]: done

File: docs/Salesforce-batchsink.md (hunk 4 above)
Comment (Shubhangi-cs) [reply docs/Salesforce-batchsink.md]: changed to Salesforce security token

File: docs/Salesforce-batchsink.md (hunk 5 above)
Comment (Shubhangi-cs) [reply docs/Salesforce-batchsink.md]: done

File: docs/Salesforce-batchsink.md (hunk 6 above)
Comment (Shubhangi-cs) [reply docs/Salesforce-batchsink.md]: done

File: docs/Salesforce-connector.md (hunk 7 above)
Comment (Shubhangi-cs) [reply docs/Salesforce-connector.md]: done

File: docs/Salesforce-streamingsource.md (hunk 8 above)
Comment (Shubhangi-cs) [reply docs/Salesforce-streamingsource.md]: done

File: docs/Salesforce-streamingsource.md (hunk 9 above)
Comment (Shubhangi-cs) [reply docs/Salesforce-streamingsource.md]: done

File: docs/Salesforce-streamingsource.md (hunk 10 above)
Comment (Shubhangi-cs) [reply docs/Salesforce-streamingsource.md]: done

File: docs/SalesforceMultiObjects-batchsource.md (hunk 11 above)
Comment (Shubhangi-cs) [reply docs/SalesforceMultiObjects-batchsource.md]: done

**Review by robinrielley — APPROVED** [review]

File: src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java [hunk 12]
```diff
@@ -129,7 +129,9 @@ public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSp
     }
     AuthenticatorCredentials authenticatorCredentials = config.getAuthenticatorCredentials();
     try {
-      SObjectDescriptor sObjectDescriptor = SObjectDescriptor.fromName(tableName, authenticatorCredentials);
+      String fields = getObjectFields(tableName);
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java:132]: What is this change for ?

File: widgets/Salesforce-batchsource.json [hunk 13]
```diff
@@ -89,6 +79,16 @@
         }
       ]
     },
+    {
+      "label": "Reference",
```
Comment (greeshmaswaminathan) [inline widgets/Salesforce-batchsource.json:83]: Is this change required ?

File: widgets/Salesforce-batchsink.json [hunk 14]
```diff
@@ -89,6 +79,16 @@
         }
       ]
     },
+    {
+      "label": "Reference",
```
Comment (greeshmaswaminathan) [inline widgets/Salesforce-batchsink.json:83]: Is this change required ?

File: widgets/Salesforce-batchsource.json (hunk 13 above)
Comment (greeshmaswaminathan) [reply widgets/Salesforce-batchsource.json:83]: Same for the other widget JSONs.

File: src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java (hunk 12 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java:132]: This change is required for the schema purpose, to limit the number of fields in schema.

File: widgets/Salesforce-batchsource.json (hunk 13 above)
Comment (Shubhangi-cs) [reply widgets/Salesforce-batchsource.json:83]: yes, this is the required change.

File: widgets/Salesforce-batchsink.json (hunk 14 above)
Comment (Shubhangi-cs) [reply widgets/Salesforce-batchsink.json:83]: yes

**Comment by vikasrathee-cs** [comment]
Closing this PR for merge conflicts. Raised a below one.
https://github.com/data-integrations/salesforce/pull/178
